### PR TITLE
Fixes synths being affected by defiler stings.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
See the title. Surr misinterpreted the concept of species and thought it was just fine to make the synth/machine species spawner subtypes immune instead. Also fixing defilers being unable to use the larval growth serum on nested hosts.

## Why It's Good For The Game
Fixing things.

## Changelog
:cl:
fix: fixes synthetics being stingable by defilers.
fix: fixing defilers being unable to use the larval growth serum on nested hosts.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/powers.dm [hunk 1]
```diff
@@ -1030,20 +1026,20 @@ GLOBAL_LIST_INIT(acid_spray_hit, typecacheof(list(/obj/structure/barricade, /obj
...
 
 /mob/living/carbon/human/can_sting()
-	if(stat != DEAD)
-		return TRUE
-	return FALSE
-
-/mob/living/carbon/human/species/machine/can_sting()
-	return FALSE
+	. = ..()
+	if(!(.))
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/powers.dm]:
```suggestion
	if(!.)
```

File: code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm [hunk 2]
```diff
@@ -61,18 +61,14 @@
...
-
 	if(!Adjacent(C))
 		if(world.time > (recent_notice + notice_delay)) //anti-notice spam
 			to_chat(src, "<span class='warning'>You can't reach this target!</span>")
 			recent_notice = world.time //anti-notice spam
 		return
 
-	if ((C.status_flags & XENO_HOST) && istype(C.buckled, /obj/structure/bed/nest))
-		to_chat(src, "<span class='warning'>Ashamed, you reconsider bullying the poor, nested host with your stinger.</span>")
+	if(!(C?.can_sting()))
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm]:
```suggestion
	if(!(C.can_sting()))
```
No need to check for `C` at this point.

File: code/modules/mob/living/carbon/xenomorph/powers.dm [hunk 3]
```diff
@@ -893,14 +893,14 @@ GLOBAL_LIST_INIT(acid_spray_hit, typecacheof(list(/obj/structure/barricade, /obj
 		M.acid_spray_act(src)
 
 
-/mob/living/carbon/Xenomorph/proc/larva_injection(mob/living/carbon/C)
-	if(!C?.can_sting())
+/mob/living/carbon/Xenomorph/proc/larva_injection(mob/living/carbon/C, precheck = TRUE)
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/xenomorph/powers.dm]: Will `precheck` ever be `TRUE`?

Comment (Ghommie) [reply code/modules/mob/living/carbon/xenomorph/powers.dm]: As of now nope, but the defiler sting runs C.can_sting(), so I didn't feel like double-checking it. But it's still around, being part of different proc.

Comment (Rohesie) [reply code/modules/mob/living/carbon/xenomorph/powers.dm]: Any reason for the extra var and check then?

**Pull request merged**